### DISCUSSION TITLE
WIP - episode duration tag / attribute

### DISCUSSION THREAD

**Discussion by dellagustin** [Podcastindex-org/podcast-namespace#488] [Ideas]
Hello all,

It is very useful for podcasts listeners if the duration of episodes can be shown at podcast players.
The RSS `enclosure` tag has an attribute for `length`, which is the size of the enclosure in bytes, but it does not support duration for media files natively.

Since the namespace also support alternate enclosures that could have a different duration, these scenario should also be considered.

### `<itunes:duration>`

From _A Podcaster’s Guide to RSS_
> The duration of an episode.
>
> Different duration formats are accepted however it is recommended to convert the length of the episode into seconds.

## References
- [RSS 2.0 enclosure tag](https://www.rssboard.org/rss-specification#ltenclosuregtSubelementOfLtitemgt)
- [Alternate Enclosure](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#alternate-enclosure) tag
- [A Podcaster’s Guide to RSS](https://help.apple.com/itc/podcasts_connect/#/itcb54353390) - at https://help.apple.com/
- [RSS feed guidelines for Google Podcasts](https://support.google.com/podcast-publishers/answer/9889544?hl=en#episode_tags)

**Comment by jamescridland** [comment]
What's the issue with `<itunes:duration>`?

**Reply by hoccau** [reply]
Hello here, and thanks for the work around Podcasting2.0 : it's amazing!
It was my understanding to that the podcast namespace could standardize the main features of podcasting experience and I didn't understand why there is not a <podcast:duration> (or something like that) in the namespace when I discovered it. If, for any reason, the itunes namespace is not wanted, I think it is missing an essential information in the podcast item.

**Comment by dellagustin** [comment]
Hello @jamescridland ,

> What's the issue with `<itunes:duration>`?

Not much, my understanding is that the podcast namespace aimed also to consolidate features from other namespaces.

One think that I don't like with `<itunes:duration>` is its lack of format definition for the value:

> Different duration formats are accepted...

Which formats? Empirically I know that hh:mm:ss is often used, but there is no clear definition. Personally it makes it easier for me as app developer if only the duration in seconds is accepted - although, I understand that the feed is also supposed to be as human readable as possible, and hh:mm:ss is human-friendlier.

Another aspect, as mentioned in the issue description:

> Since the namespace also supports alternate enclosures that could have a different duration, these scenario should also be considered.

This is something where `<itunes:duration>` would not help, as it is only supposed to be used directly under the `<item>`.

**Comment by theDanielJLewis** [comment]
I really don't think we should limit ourselves based on making the RSS feed "human-readable." It's a coding style intended to be read by apps, not humans. And it's already rather unreadable simply because of the existence of tags.
